sled-agent/src/services.rs

```diff
@@ -3831,6 +3843,33 @@ impl ServiceManager {
         }
     }
 
+    /// Check if the system time is synchronized and if so, execute the
+    /// boottime_rewrite function and send messages.
+    async fn if_timesynced(&self) {
```

Comment:
The boottime rewrite effect was somewhat hidden on the sync check; so, `if_timesynced` now exposes any type of events we may want to do once we make the switch on `compare_exchange`.